From b39b40b9bb103024200c877f83bffc389ce0c29b Mon Sep 17 00:00:00 2001
From: Axel Huebl <axel.huebl@plasma.ninja>
Date: Fri, 5 Mar 2021 17:54:47 -0800
Subject: [PATCH 2/4] Complex: Support OpenBLAS w/ MSVC

MSVC does not have C99 Complex.

At least support OpenBLAS' complex structs.
---
 include/blas/config.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/include/blas/config.h b/include/blas/config.h
index d6d44ab..e3a2db3 100644
--- a/include/blas/config.h
+++ b/include/blas/config.h
@@ -30,6 +30,10 @@
     #include <complex>
     typedef std::complex<float>  blas_complex_float;
     typedef std::complex<double> blas_complex_double;
+#elif defined(_MSC_VER)
+    /* MSVC has no C99 _Complex */
+    typedef struct { float real, imag; }  blas_complex_float;
+    typedef struct { double real, imag; } blas_complex_double;
 #else
     /* otherwise, by default use C99 _Complex */
     #include <complex.h>
-- 
2.25.1

